impl/integration/env.go

```diff
@@ -159,13 +160,19 @@ func NewEnv() (*Env, error) {
 	if err != nil {
 		return nil, fmt.Errorf("env: Failed to create mutations object: %v", err)
 	}
-	auth := authentication.NewFake()
+	authFunc := authentication.FakeAuthFunc
 	authz := authorization.New()
 
 	queue := mutator.MutationQueue(mutations)
 	server := keyserver.New(logEnv.Log, mapEnv.Map, logEnv.Admin, mapEnv.Admin,
-		entry.New(), auth, authz, domainStorage, queue, mutations)
-	gsvr := grpc.NewServer()
+		entry.New(), authz, domainStorage, queue, mutations)
+	gsvr := grpc.NewServer(
+		grpc.UnaryInterceptor(
+			authentication.UnaryServerInterceptor(map[string]grpc_auth.AuthFunc{
+				"/google.keytransparency.v1.KeyTransparency/UpdateEntry": authFunc,
```

Comment:
Deja vu.

Reply:
The integration tests act like a sort of "main", wiring together libraries much like main does. 
As a result, this intentionally lives in `impl/` and allows other environments to setup their own custom `envs`

Reply:
OK if it doesn't rely on those strings matching (my earlier point) then it might not be a problem.

Reply:
I don't see a place in the generated proto files where these strings are publicly exported.  Not sure what the best solution is.